pkg/controller/backup_controller.go

```diff
@@ -958,30 +957,75 @@ func (b *backupReconciler) waitVolumeSnapshotReadyToUse(ctx context.Context,
 	vsChannel := make(chan snapshotv1api.VolumeSnapshot, len(volumeSnapshots))
 	defer close(vsChannel)
 
+	type ErrorAndReadyToUse struct {
+		err        error
+		readyToUse bool
+	}
+
+	vsStatusMap := make(map[string]ErrorAndReadyToUse)
+
 	for index := range volumeSnapshots {
-		volumeSnapshot := volumeSnapshots[index]
-		eg.Go(func() error {
-			err := wait.PollImmediate(interval, timeout, func() (bool, error) {
-				tmpVS, err := b.volumeSnapshotClient.SnapshotV1().VolumeSnapshots(volumeSnapshot.Namespace).Get(b.ctx, volumeSnapshot.Name, metav1.GetOptions{})
-				if err != nil {
-					return false, errors.Wrapf(err, fmt.Sprintf("failed to get volumesnapshot %s/%s", volumeSnapshot.Namespace, volumeSnapshot.Name))
+		vsStatusMap[volumeSnapshots[index].Name] = ErrorAndReadyToUse{err: nil, readyToUse: false}
+	}
+	eg.Go(func() error {
```

Comment:
This is the eventual polling which is done post backup flow for populating VSC.json
We can and should actually remove any Polling done here completely
Or switch to the parallel route I have covered here.